src/shadowbox/model/access_key.ts

```diff
@@ -18,25 +18,37 @@ export type AccessKeyMetricsId = string;
 // Parameters needed to access a Shadowsocks proxy.
 export interface ProxyParams {
   // Hostname of the proxy
-  hostname: string;
+  readonly hostname: string;
   // Number of the port where the Shadowsocks service is running.
-  portNumber: number;
+  readonly portNumber: number;
   // The Shadowsocks encryption method being used.
-  encryptionMethod: string;
+  readonly encryptionMethod: string;
   // The password for the encryption.
-  password: string;
+  readonly password: string;
+}
+
+// Parameters needed to enforce an access key data transfer quota, over a sliding window.
+export interface AccessKeyQuota {
+  // The allowed metered data transfer measured in bytes.
+  readonly quota: {bytes: number};
+  // The sliding window size in hours.
+  readonly window: {hours: number};
 }
 
 // AccessKey is what admins work with. It gives ProxyParams a name and identity.
 export interface AccessKey {
   // The unique identifier for this access key.
-  id: AccessKeyId;
+  readonly id: AccessKeyId;
   // Admin-controlled, editable name for this access key.
-  name: string;
+  readonly name: string;
   // Used in metrics reporting to decouple from the real id. Can change.
-  metricsId: AccessKeyMetricsId;
+  readonly metricsId: AccessKeyMetricsId;
   // Parameters to access the proxy
-  proxyParams: ProxyParams;
+  readonly proxyParams: ProxyParams;
+  // Admin-controlled, data transfer quota for this access key. Unlimited if unset.
```

Comment:
Replace "Ulimited if unset" with "Server's default quota (unlimited by default) if unset"?

Reply:
I'll update the comment and other documentation when I implement default quotas.